tests/tests/test-gas-estimation2.ts

```diff
@@ -0,0 +1,18 @@
+import { expect } from "chai";
+
+import { describeDevMoonbeam } from "../util/setup-dev-tests";
+
+import { createContract } from "../util/transactions";
+
+describeDevMoonbeam("Estimate Gas - infinite loop", (context) => {
+  it("Should be able to estimate gas of infinite loop call", async function () {
+    const { contract, rawTx } = await createContract(context.web3, "InfiniteContract");
+    await context.createBlock({ transactions: [rawTx] });
+
+    expect(
+      await contract.methods.infinite().estimateGas({
+        gas: null,
+      })
+    ).to.equal(15_000_000);
```

Comment:
Shouldn't an infinite fail with out of gas instead of returning a value ?

Reply:
I think this test is ensuring that when we call `estimate_gas` without specifying gas that it should (1) use our block-gas-limit instead and (2) consume all of it when we OOG.

Reply:
This behavior is due to the way the binary search algo works. If the transaction returns an OOG in any case, the maximum amount of gas is returned.